examples/amm/src/contract.rs

```diff
@@ -446,9 +446,8 @@ impl AmmContract {
 
     fn get_message_creation_chain_id(&mut self) -> ChainId {
         self.runtime
-            .message_id()
-            .expect("Getting message id should not fail")
-            .chain_id
+            .message_origin_chain_id()
+            .expect("Getting message origin chain ID should not fail")
     }
```

Comment:
That code section should be at the beginning of the `execute_message`.

Reply:
Done in https://github.com/linera-io/linera-protocol/pull/4289/commits/dafcaf52e91e62111d2b95a26a5fcc5a7475e19c.